tst/testinstall/MatrixObj/testmatobj_new.g

```diff
@@ -0,0 +1,422 @@
+            ErrorNoReturn(Concatenation("Error while generating examples. In a case with NrRows = ", String(ex[2], " NrCols = ", String(ex[3]), " domain = ", String(ex[4]), "and filter := ", String(filter), " NumberRows, NumberColumns, BaseDomain or ConstructingFilter does not work.")));
+        fi;
+    od;
+
+    return result;
+end;
+
+# This is a concrete test function. It gets an example and should test the
+# function it specifies in its identifier. Here the function tests MatElm that
+# is element access.
+MatObjTest_TestMatElm := function(ex)
+    local col, row, elm;
+
+    for row in [1..NrRows(ex.matObj)] do
+        for col in [1..NrCols(ex.matObj)] do
+            elm := ex.matObj[row, col];
+            if elm <> ex.sourceOfTruth[row,col] then
+                return false;
+            fi; 
+        od;
+    od;
+
+    return true;
+end;
+
+MatObjTest_TestSetMatElm := function(ex_in)
+    local col, row, elm, mat, ex;
+
+    ex := ShallowCopy(ex_in);
+    mat := ex.matObj;
+    elm := Zero(BaseDomain(mat));
+    ex.sourceOfTruth[row, col] := elm;
+    ex.mat[row, col] := elm;
+
+    for row in [1..NrRows(ex.matObj)] do
+        for col in [1..NrCols(ex.matObj)] do
+            ex.matObj[row,col] := elm;
+            ex.sourceOfTruth[row,col] := elm;
+            if ex.matObj[row,col] <> ex.sourceOfTruth[row,col] then
+                return false;
+            fi; 
+        od;
+    od;
+
+    return true;
+end;
+
+MatObjTest_TestIsIdentityMat := function(ex_in)
+    #todo
+end;
+
+MatObjTest_TestInverse := function(ex_in)
+    local ex, mat, inv;
+
+    ex := ShallowCopy(ex_in);
+    mat := ex.MatObj;
+    inv := mat^(-1);
+    if IsIdentityMat(mat*inv) and IsIdentityMat(inv*mat) then 
+        return true;
+    fi;
+
+    return false;
+end;
+
+# This function requires a filter defining a MatrixObject and a record of 
+# options. 
+# This function runs tests to check whether the specified MatrixObject
+# implementation implements required methods and if these and other methods are 
+# correct (in the sense that they give correct results for the cases tested 
+# here).
+# The following options are supported:
+# randomSource:
+#   See TestMatrixObj_GenerateExample
+# state:
+#   See TestMatrixObj_GenerateExample
+# dimensions: 
+#   See TestMatrixObj_GenerateExample
+# dimensions_red: A list of tuples defining dimensions for matrices. If not set
+#   this is set to [[1,1],[2,2],[3,3],[5,5]]. These dimensions are used to
+#   generate examples for more expensive tests. 
+# domains:
+#   See TestMatrixObj_GenerateExample
+# domains_red:
+#   A list of domains in which matrices should be generated. By default
+#   this is set to [Integers, Rationals, GF(2), GF(5)]. These domains are used 
+#   to generate examples for more expensive tests.
```

Comment:
Based on what you write elsewhere, I assume "red" stands for "reduced" and not the color "red"? But why "reduced" ?

It might also be good to explain how this relates to `domains`. E.g. is this right? "If domains_red is given, then it is used instead of domains for certain expensive tests.

Reply:
I chose 'reduced' since its an option to choose a smaller set of cases to test. There might be a better name though.